### PR TITLE
bump sdk 47, ibc v7, wasmd 40

### PR DESCRIPTION
**Pull request by cgsingh33**
summary:

Done:
sdk bump 0.47.5
wasmd 0.41
cometbft 0.37.2
ibc-go v7.2
update protogen script
migration code (needs review)

Removed:
ibcratelimit

### CODE REVIEW

File: types/utils.go [hunk 1]
```diff
@@ -144,7 +144,8 @@ func RandomDec(r *rand.Rand, min, max sdk.Dec) sdk.Dec {
 // GenAndDeliverTx generates a transactions and delivers it.
 func GenAndDeliverTx(txCtx simulation.OperationInput, fees sdk.Coins, gas uint64) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
 	account := txCtx.AccountKeeper.GetAccount(txCtx.Context, txCtx.SimAccount.Address)
-	tx, err := helpers.GenTx(
+	tx, err := simtestutil.GenSignedMockTx(
+		rand.New(rand.NewSource(time.Now().UnixNano())),
```
Comment (dheerajkd30) [inline types/utils.go:149]: Why do we need rand ?

Comment (cgsingh33) [reply types/utils.go:149]: "math/rand" is only used in tests and simulations files throughout the codebase, and so is this function.

File: x/lend/client/cli/tx.go [hunk 2]
```diff
@@ -1374,7 +1402,8 @@ func CmdDepreciatePoolsProposal() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			txf := tx.NewFactoryCLI(clientCtx, cmd.Flags()).WithTxConfig(clientCtx.TxConfig).WithAccountRetriever(clientCtx.AccountRetriever)
+			txf, err := tx.NewFactoryCLI(clientCtx, cmd.Flags())
```
Comment (dheerajkd30) [inline x/lend/client/cli/tx.go:1405]: err handling missing here.

File: x/lend/client/cli/tx.go [hunk 3]
```diff
@@ -1432,7 +1461,8 @@ func CmdAddEModePairsProposal() *cobra.Command {
...
```
Comment (dheerajkd30) [inline x/lend/client/cli/tx.go:1467]: err handling missing here.

File: app/app.go [hunk 4]
```diff
@@ -602,19 +620,63 @@ func New(
 	)
 	app.IbcHooksKeeper = &hooksKeeper
 
-	app.WireICS20PreWasmKeeper(appCodec, baseApp, app.IbcHooksKeeper, scopedTransferKeeper)
+	cmdxPrefix := sdk.GetConfig().GetBech32AccountAddrPrefix()
```
Comment (dheerajkd30) [inline app/app.go:623]: checkpoint + revisit

Comment (cgsingh33) [reply app/app.go:623]: Works as it gives 'comdex' only

**Review by pratikasr — APPROVED** [review]

**Review by dheerajkd30 — APPROVED** [review]

**Pull request merged**